front-project/src/features-store/auth/auth.thunks.js

```diff
@@ -0,0 +1,35 @@
+import { authApi } from "../../api/auth.api";
+import {
+  setAuthErr,
+  setAuthFinished,
+  setAuthStarted,
+  signIn,
+  signUp,
+} from "./auth.actions";
+
+export const signUpUserThunk = ({ username, email, password }) => {
+  return async function (dispatch) {
+    dispatch(setAuthStarted());
+    authApi
+      .signUp({ username, email, password })
+      .then((res) => dispatch(signUp(res.data.user)))
+      .then(() => dispatch(setAuthFinished()))
+      .catch((err) => {
+        console.log(err);
+        dispatch(setAuthErr(JSON.parse(err).message));
```

Comment:
see example in previous comments 
`
return { [FORM_ERROR]: err.message || 'Login error' }
`